### PR TITLE
Fix incorrect rounding of tablet aspect ratio values

### PR DESCRIPTION
**Pull request by swekka**
The values of the aspect ratio on the tablet area preview are sometimes incorrectly rounded down.
For example: My area here is set to 80x45, but due to `sizeX`, `sizeY` and their `SettingsSlider`'s being floats the values are incorrectly rounded down.
image
I also changed the functions from my previous PR to return floats, since I don't see the point in converting them to integers when `sizeX` and `sizeY` are floats.

### CODE REVIEW

File: osu.Game/Overlays/Settings/Sections/Input/TabletAreaSelection.cs [hunk 1]
```diff
@@ -212,17 +212,17 @@ protected override void Update()
         {
             base.Update();
 
-            if (!(tablet.Value?.Size is Vector2 size))
-                return;
```
Comment (bdach) [inline osu.Game/Overlays/Settings/Sections/Input/TabletAreaSelection.cs:216]: I prefer the old version of the code here. We generally prefer early-returns to nested `if`s.

File: osu.Game/Overlays/Settings/Sections/Input/TabletSettings.cs [hunk 2]
```diff
@@ -342,8 +342,8 @@ private void forceAspectRatio(float aspectRatio)
 
         private float currentAspectRatio => sizeX.Value / sizeY.Value;
 
-        private static int getHeight(float width, float aspectRatio) => (int)Math.Round(width / aspectRatio);
+        private static float getHeight(float width, float aspectRatio) => MathF.Round(width / aspectRatio);
 
-        private static int getWidth(float height, float aspectRatio) => (int)Math.Round(height * aspectRatio);
+        private static float getWidth(float height, float aspectRatio) => MathF.Round(height * aspectRatio);
```
Comment (bdach) [inline osu.Game/Overlays/Settings/Sections/Input/TabletSettings.cs:347]: Would prefer the type here to stay `int` too, I think? BIt weird for me to change this return type given the fact that the returned number is a whole number anyway.

Comment (Samaoo) [reply osu.Game/Overlays/Settings/Sections/Input/TabletSettings.cs]: That was a mistake, I didn't mean to round it. I changed it now.

**Comment by bdach** [comment]
Code looks reasonable but I'd like to put this through device testing before final approval, and I don't have one on me right now. Unless someone else tests this may need to wait a few days.

I'm slightly suspicious of the change in `TabletSettings` - will need to inspect how the setting sliders behave after forcing aspect ration with this change. The thinking is that it may start displaying a funky value in the tooltip and the bindables may need to become `BindableFloat`s.

**Comment by Samaoo** [comment]
> I'm slightly suspicious of the change in `TabletSettings` - will need to inspect how the setting sliders behave after forcing aspect ration with this change.

The reason I made this change is because when changing `sizeX` and `sizeY` with sliders, a decimal number is assigned to them, but when changing them by forcing/applying an aspect ratio it calls `getWidth()` and `getHeight()` and a whole number is assigned to them. I think that it should either always be a decimal number or always be rounded to a whole number.

This is unrelated but I think it's weird to not show decimal places when `sizeX` and `sizeY` are decimal numbers.

**Comment by bdach** [comment]
> This is unrelated but I think it's weird to not show decimal places when `sizeX` and `sizeY` are decimal numbers.

Decimal places can be shown, I just want to check how many are actually displayed. Floats have a tendency to have really ugly decimal expansions and I wouldn't want to show stuff like 166.6666666667 or the like.

**Comment by Samaoo** [comment]
> Decimal places can be shown, I just want to check how many are actually displayed. Floats have a tendency to have really ugly decimal expansions and I wouldn't want to show stuff like 166.6666666667 or the like.

I agree, that can easily be fixed by changing
```csharp
private readonly BindableNumber<float> sizeX = new BindableNumber<float> { MinValue = 10};
private readonly BindableNumber<float> sizeY = new BindableNumber<float> { MinValue = 10};
```
to
```csharp
private readonly BindableNumber<float> sizeX = new BindableNumber<float> { MinValue = 10, Precision = 0.01f };
private readonly BindableNumber<float> sizeY = new BindableNumber<float> { MinValue = 10, Precision = 0.01f };
```
but if the PR is merged as-is, it wont show stuff like 166.6666666667 because the default precision is 1.

**Review by bdach — APPROVED** [review]

**Comment by bdach** [comment]
Seems to be behaving fine in device testing.

**Pull request merged**